WordPress/Classes/ViewRelated/Support/LogOutActionHandler.swift

```diff
@@ -0,0 +1,38 @@
+import UIKit
+
+struct LogOutActionHandler {
+
+    func logOut(with viewController: UIViewController) {
+        let alert  = UIAlertController(title: logOutAlertTitle, message: nil, preferredStyle: .alert)
+        alert.addActionWithTitle(Strings.alertCancelAction, style: .cancel)
+        alert.addActionWithTitle(Strings.alertLogoutAction, style: .destructive) { [weak viewController] _ in
+            viewController?.dismiss(animated: true) {
+                AccountHelper.logOutDefaultWordPressComAccount()
+            }
+        }
+        viewController.present(alert, animated: true)
+    }
+
+    private var logOutAlertTitle: String {
+        let context = ContextManager.sharedInstance().mainContext
+        let count = AbstractPost.countLocalPosts(in: context)
+
+        guard count > 0 else {
+            return Strings.alertDefaultTitle
+        }
+
+        let format = count > 1 ? Strings.alertUnsavedTitlePlural : Strings.alertUnsavedTitleSingular
+        return String(format: format, count)
+    }
+
+
+    private struct Strings {
+        static let alertDefaultTitle = AppConstants.Logout.alertTitle
+        static let alertUnsavedTitleSingular = NSLocalizedString("You have changes to %d post that hasn't been uploaded to your site. Logging out now will delete those changes. Log out anyway?",
+                                                            comment: "Warning displayed before logging out. The %d placeholder will contain the number of local posts (SINGULAR!)")
+        static let alertUnsavedTitlePlural = NSLocalizedString("You have changes to %d posts that haven’t been uploaded to your site. Logging out now will delete those changes. Log out anyway?",
+                                                          comment: "Warning displayed before logging out. The %d placeholder will contain the number of local posts (PLURAL!)")
+        static let alertCancelAction = NSLocalizedString("Cancel", comment: "Verb. A button title. Tapping cancels an action.")
+        static let alertLogoutAction = NSLocalizedString("Log Out", comment: "Button for confirming logging out from WordPress.com account")
+    }
```

Comment:
Should we use the new reverse dns notation here?

Reply:
Since the log out logic is already implemented in `MeViewController`, I re-used the same logic for the support screen, so those strings aren't new. I think if we change the keys, then translators will have to provide a new translations for those strings, right? If so, is there way to change the keys without losing the existing translation?

Reply:
Thanks for the ping!

So:
 - Using the exact same strings as the `MeViewController` is fine, as long as you are really sure that they are used in the same kind of UI context (and that they should thus have the exact same translations in both places, because they really are about the exact same thing)
    - That being said, instead of copy/pasting the strings and comments in there, and thus duplicating the declaration of those `NSLocalizedStrings` (between the ones in the `MeViewController` and the ones here) — which could lead to a later PR changing the comment here, but not in the other place, and lead to more confusion / discrepancies by not having a single source of truth for those code declarations), I'd instead suggest to reference the existing constants from `MeViewController` (or move those constants into a common class / file / ViewModel / what have you, as long as you declare them only once and reference them from both places)
    - This might also be a more "dangerous" assumption to make for shorter copies (like "Cancel" and "Log Out") which seems like words that are small and common enough to be used in many other places in the app. So if another place in the app has `NSLocalizedString("Cancel", comment: "Some completely different context")`, maybe the translators would be tempted to translate it differently for one context than for the other, but if you use the same key for too many different contexts and places in the code, that might lead to translation inconsistencies, [like what happened here in the past](https://github.com/wordpress-mobile/WordPress-iOS/pull/19136).

The last point and example is one more key reason why semantic keys are helpful and that it could still be a good idea to switch to them here 😉 

> If so, is there way to change the keys without losing the existing translation?

Not automatically… but almost.

It happens that GlotPress has a system they call "translation memory", and when it detects a source (English) copy for which they already had a translation for in the past, even if for a different key,… then it is automatically suggested to translators:

<img width="866" alt="image" src="https://user-images.githubusercontent.com/216089/202778521-22a2d3c5-78cc-4200-b509-378665f7b0a6.png">
_See "Suggestions from Translation Memory" above_

This means that, yes, if you change the key in this `NSLocalizedString` call to start using semantic keys here instead, the translation for that new key will start as empty / untranslated the first time that new key will be imported into GlotPress… but then it will be only one click away for polyglots to translate (picking if from the suggested translation from Translation Memory) as opposed to having to re-translate it from scratch. So that new key will get the existing translation from the old key pretty quickly, so I'd say in the end that's not that big a deal.

Reply:
Thanks for clarifying @AliSoftware 🙇 

>I'd instead suggest to reference the existing constants from MeViewController (or move those constants into a common class / file / ViewModel / what have you, as long as you declare them only once and reference them from both places)

I totally agree with this and the log out logic duplication is temporary. Right now the type `LogOutActionHandler` is used only in `SupportTableViewController` but I'll open another PR so that `MeViewController` uses it as well. As a result, those strings declaration will no longer be duplicated.